xdsl/interactive/app.py

```diff
@@ -79,23 +89,38 @@ def compose(self) -> ComposeResult:
                     yield Button("Copy Output", id="copy_output_button")
         yield Footer()
 
+    @on(SelectionList.SelectedChanged)
+    def update_selected_view(self) -> None:
+        """
+        When the SelectionList (pass options) changes (i.e. a pass was selected or deselected), update the label to show
+        the query, and then call the update_current_module() function, which applies the selected passes to the input and displays the output
+        """
+        self.update_current_module()
+
     @on(TextArea.Changed, "#input")
```

Comment:
```suggestion
    @on(SelectionList.SelectedChanged)
    @on(TextArea.Changed, "#input")
```

You can stack the `@on` annotations it just means that when either of those events are received this method will be called